### PR TITLE
Use structured suggestion for unnecessary bounds in type aliases

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

File: src/test/ui/associated-type-bounds/type-alias.stderr [hunk 1]
```diff
@@ -5,93 +5,129 @@ LL | type _TaWhere1<T> where T: Iterator<Item: Copy> = T;
    |                         ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(type_alias_bounds)]` on by default
-   = help: the clause will not be checked when the type alias is used, and should be removed
+help: the clause will not be checked when the type alias is used, and should be removed
+   |
+LL | type _TaWhere1<T>  = T;
```
Comment (davidtwco) [inline src/test/ui/associated-type-bounds/type-alias.stderr]: Do you think it would be viable to extend the span so that we don't leave two spaces behind?

**Review by davidtwco — APPROVED** [review]
LGTM, left a suggestion, r=me if you don't think it's something we could do.

File: src/test/ui/associated-type-bounds/type-alias.stderr (hunk 1 above)
Comment (estebank) [reply src/test/ui/associated-type-bounds/type-alias.stderr]: I noticed that too. The span wrangling needed to fix that makes me uneasy. I'd say let's leave this as is for now.

**Pull request merged**